vmc/resource_site_recovery.go

```diff
@@ -0,0 +1,183 @@
+			},
+			"site_recovery_state": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Site recovery state. Possible values are: ACTIVATED, ACTIVATING, CANCELED, DEACTIVATED, DEACTIVATING, DELETED, FAILED",
+			},
+			"user_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "User id that last updated this record.",
+			},
+			"user_name": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "User name that last updated this record.",
+			},
+			"srm_nodes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeMap},
+			},
+			"vr_node": {
+				Type:     schema.TypeMap,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceSiteRecoveryCreate(d *schema.ResourceData, m interface{}) error {
+
+	connector := (m.(*ConnectorWrapper)).Connector
+
+	siteRecoveryClient := draas.NewDefaultSiteRecoveryClient(connector)
+
+	srmExtensionKeySuffix := d.Get("srm_extension_key_suffix").(string)
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	sddcID := d.Get("sddc_id").(string)
+
+	activateSiteRecoveryConfigParam := &model.ActivateSiteRecoveryConfig{
+		SrmExtensionKeySuffix: &srmExtensionKeySuffix,
+	}
+
+	task, err := siteRecoveryClient.Post(orgID, sddcID, activateSiteRecoveryConfigParam)
+
+	if err != nil {
+		return fmt.Errorf("Error while activating site recovery for sddc %s: %v", sddcID, err)
+	}
+
+	// Wait until site recover is activated
+	taskID := task.ResourceId
+	d.SetId(*taskID)
+	return resource.Retry(d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
+		tasksClient := draas.NewDefaultTaskClient(connector)
+		task, err := tasksClient.Get(orgID, task.Id)
+		if err != nil {
+			if err.Error() == (errors.Unauthenticated{}.Error()) {
+				log.Print("Auth error", err.Error(), errors.Unauthenticated{}.Error())
+				err = (m.(*ConnectorWrapper)).authenticate()
+				if err != nil {
+					return resource.NonRetryableError(fmt.Errorf("Error authenticating in CSP: %s", err))
+				}
+				return resource.RetryableError(fmt.Errorf("Instance creation still in progress"))
+			}
+			return resource.NonRetryableError(fmt.Errorf("Error describing instance: %s", err))
+
+		}
+		if *task.Status != "FINISHED" {
+			return resource.RetryableError(fmt.Errorf("Expected instance to be created but was in state %s", *task.Status))
+		}
+		return resource.NonRetryableError(resourceSiteRecoveryRead(d, m))
+	})
+	return nil
+
+}
+
+func resourceSiteRecoveryRead(d *schema.ResourceData, m interface{}) error {
+	connector := (m.(*ConnectorWrapper)).Connector
+	sddcID := d.Id()
+	orgID := (m.(*ConnectorWrapper)).OrgID
+	siteRecoveryClient := draas.NewDefaultSiteRecoveryClient(connector)
+	siteRecovery, err := siteRecoveryClient.Get(orgID, sddcID)
+	if err != nil {
```

Comment:
do we plan re-create resource if it is not found? Eg: case when its deleted via console accidentally